server/src/main/java/org/elasticsearch/action/admin/indices/refresh/TransportUnpromotableShardRefreshAction.java

```diff
@@ -11,37 +11,44 @@
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.ActionResponse;
 import org.elasticsearch.action.support.ActionFilters;
-import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.action.support.broadcast.unpromotable.TransportBroadcastUnpromotableAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.service.ClusterService;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.index.engine.Engine;
 import org.elasticsearch.index.shard.IndexShard;
 import org.elasticsearch.indices.IndicesService;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportService;
 
-public class TransportUnpromotableShardRefreshAction extends HandledTransportAction<UnpromotableShardRefreshRequest, ActionResponse.Empty> {
-    public static final String NAME = RefreshAction.NAME + "[u]";
+public class TransportUnpromotableShardRefreshAction extends TransportBroadcastUnpromotableAction<UnpromotableShardRefreshRequest> {
+
+    public static final String NAME = RefreshAction.NAME + "/unpromotable";
```

Comment:
I have a doubt so I prefer to comment: does this new action require specific privileges? IIRC it is ok as long as the original action has been authorized but I'd be happy if someone can confirm because I see `indices:admin/refresh[s]` and `indices:admin/refresh*` privileges in the codebase.

Reply:
Hi @tlrx ! By privileges, what do you mean? Do you mean the `x-pack/plugin/security/qa/operator-privileges-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/operator/Constants.java` file? We did not need to put it there, because of [this comment](https://github.com/elastic/elasticsearch/pull/93600#discussion_r1100488177) whereby now this new action is internal and not exposed to clients I believe.

Reply:
I think I can rename the action to what we have currently in `main`, i.e., `indices:admin/refresh[s][u]`. or maybe better `indices:admin/refresh[u]`. This followed a similar approach to how the replication messages were done, that append a `[r]` to the action name. Let me see if that resolves the CI issues.

Reply:
The `RBACEngine#authorizeIndexAction` engine is doing a bunch of verifications depending on the action name (that starts with `indices:`) and if the UnpromotableShardRefreshRequest is an instance or not of `IndicesRequest`. 

This is not the case so the only known actions that contains `indices:` and are not `IndicesRequest` are scroll and async search it seems.

That being said we must be compliant with the RBACEngine or adjust its checks, but I don't think we need new privileges.

Reply:
I guess I got confused because currently in `main` we do call the same action (see [here](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/indices/refresh/TransportShardRefreshAction.java#L110-L120)) without an RBAC error. So I am not sure why this issue was created now.

I just pushed a fix for RBACEngine rules. Feel free to review as I am not aware of that code piece.

Reply:
It used `transportService.sendChildRequest()` which is `transportService.sendRequest()` in this PR

Reply:
I tried that also but it did not work locally. Are you sure that was the difference? Because i can try that again if that's preferred?

Reply:
I will resolve this conversation, but feel free to unresolve it during the main reviewing or comment in the part with the RBACEngine changes!